aws/signer/v4/v4.go

```diff
@@ -407,9 +414,25 @@ var SignRequestHandler = request.NamedHandler{
 // If the credentials of the request's config are set to
 // credentials.AnonymousCredentials the request will not be signed.
 func SignSDKRequest(req *request.Request) {
-	signSDKRequestWithCurrTime(req, time.Now)
+	signSDKRequestWithCurrTime(req, time.Now, Config{})
+}
+
+// Config is used to dictate how to sign requets.
+type Config struct {
+	UnsignedBody bool
+}
+
+// BuildNamedHandler will build a generic signer
+func BuildNamedHandler(name string, cfg Config) request.NamedHandler {
```

Comment:
Instead of taking a cfg struct, i'd suggest functional options that modify `Signer` directly setting the options they want. This unifies the SDK's services configuration options and using the Signer standalone. This also simplifies the changes that would be needed to configure single vs double encoding for services like S3 instead of hardcoded here in `signSDKRequestWithCurrTime`.

e.g
```go
// Can a Option type be created?  Might conflict with NewSigner. If not all the Option below would need to be replaced with func(*Signer).
type Option func(*Signer)

func (s *Signer) ApplyOptions(opts ...[]Option) {
    for _, opt := range opts {
        opt(s)
    }
}

func WithUnsignedBody(b bool) Option {
    return func(s *Signer) {
        s.UnsignedBody = b
    }
}

func withTimeNow() Option {
   return func(s *Signer) {
        s. currentTimeFn = time.Now
   }
}

func BuildNamedHandler(name string, opts ...[]Option) request.NamedHandler {
    // ...
   signSDKRequestWithCurrTime(req, append([]Option{withTimeNow}, opts...))
   // ...
}

func signSDKRequestWithCurrTime(req *request.Request, opts ...[]Option) {
    // ...
   // Add SDK service API op customizations to signer via appending to func options.
   // TODO: Maybe invert with if service API op customizations are first or last in list.
   opts = append(opts, func(v4 *Signer) {
       // ...
   })

   v4 := NewSigner(req.Config.Credentials, opts...)
   // ...
}
```

This also allows the curTimeFn to be moved to an unexported `Option` also.